contrib/vagrant/start.sh

```diff
@@ -321,11 +321,9 @@ function write_cilium_cfg() {
 
     if [ -n "${K8S}" ]; then
         cilium_options+=" --k8s-kubeconfig-path /var/lib/cilium/cilium.kubeconfig"
-        cilium_options+=" --kvstore etcd"
-        cilium_options+=" --kvstore-opt etcd.config=/var/lib/cilium/etcd-config.yml"
+        cilium_options+=" --identity-allocation-mode=crd --enable-k8s-event-handover=false"
```

Comment:
can we have `cilium_options_with_kvstore` which is `cilium_options_with_kvstore="${cilium_options} --kvstore etcd --kvstore-opt etcd.config=/var/lib/cilium/etcd-config.yml"` and then have it on line 347 `echo 'CILIUM_OPTS="${cilium_options_with_kvstore}"' >> /etc/sysconfig/cilium`? Same for the operator.

Reply:
I'm not sure I understand what you're proposing. I'm trying to disable kvstore here.

Reply:
I think Andre is proposing to have a separate variable containing the Cilium options with and without the kvstore options. Then depending on if `K8S` is provided, we use the one without kvstore opts. That separate variable containing the kvstore opts though is still written out to the sysconfig file, so that we may use it if we want to turn on kvstore. It'll help situations where we aren't just stuck with the opts that the `K8S=1` VM was provisioned with.

Reply:
Correct. We can have as many as `CILIUM_OPTS` we want, only the last one is considered to be used when starting the agent.

Reply:
Sorry, but I actually still don't get it. Note the kvstore options are currently different if `K8S=1` or not, even though kvstore is always enabled. So if I define a variable with `--kvstore etcd --kvstore-opt etcd.config=/var/lib/cilium/etcd-config.yml`, it will never be used.

If we want to remember how to quickly enable kvstore when `K8S=1`, maybe I can just add a comment with `--kvstore etcd --kvstore-opt etcd.config=/var/lib/cilium/etcd-config.yml`?

Reply:
Define `CILIUM_OPTS` twice and store different opts depending on `K8S`. Note, the last `CILIUM_OPTS` will be the one that takes effect because it overrides the previous one.

```if K8S == 1:
  CILIUM_OPTS=(all kvstore opts)
  CILIUM_OPTS=(the changes that you have in this PR)
else:
  CILIUM_OPTS=(the changes that you have in this PR)
  CILIUM_OPTS=(all kvstore opts)
```

Something like that, @aanm to confirm

Reply:
yes

Reply:
I updated the branch.